### PR TITLE
Maksym/feature/signup

### PR DESCRIPTION
**Pull request by MaksymKuzmych**

### CODE REVIEW

File: services/users.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import axios from 'axios';
+
+import { UserIn } from '@/client';
+
+export const getAllUsers = async (uuid: string) => {
```
Comment (Taraslysak) [inline services/users.ts:5]: I suggest we pass session to these endpoints as argument. Thus we will be able to make a reusable functions to process session here

File: components/NewUserForm/NewUserForm.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    const formData = new FormData(event.currentTarget);
+
+    const username = formData.get('username') as string;
+    const email = formData.get('email') as string;
+    const password = formData.get('password') as string;
+    const confirmPassword = formData.get('confirmPassword') as string;
+    const role = formData.get('role') as UserRole;
+
+    if (password === confirmPassword) {
```
Comment (Taraslysak) [inline components/NewUserForm/NewUserForm.tsx:28]: and if password and confirm password are empty strings?

File: components/Navigation/Navigation.constants.ts [hunk 3]
```diff
@@ -1,7 +1,13 @@
-export const NAV_LINKS = [
-  { label: 'home', href: '/home', isProtected: false },
-  { label: 'properties', href: '/properties', isProtected: false },
-  { label: 'about', href: '/about', isProtected: false },
-  { label: 'terms', href: '/terms', isProtected: false },
-  { label: 'admin', href: '/admin', isProtected: true },
+export const PUBLIC_NAV_LINKS = [
```
Comment (Taraslysak) [inline components/Navigation/Navigation.constants.ts:1]: may be create `ILink` interface of some sort for these arrays?

File: app/api/users/route.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    return NextResponse.json(users);
+  } catch (error) {
+    return NextResponse.next();
+  }
+}
+
+export async function POST(request: Request) {
+  const { searchParams } = new URL(request.url);
+  const uuid = searchParams.get('uuid') || '';
```
Comment (Taraslysak) [inline app/api/users/route.tsx:20]: I think we can make util function for this code

**Review by Taraslysak — COMMENTED** [review]
Please consider some refactoring

File: components/NewUserForm/NewUserForm.tsx (hunk 2 above)
Comment (MaksymKuzmych) [reply components/NewUserForm/NewUserForm.tsx:28]: i will replace it to formik and handle all this cases

File: app/api/api.utils.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export const getAuthUuid = (request: Request) => {
```
Comment (Taraslysak) [inline app/api/api.utils.ts:1]: `export const getAuthUuid = (request: Request): string =>`

File: app/api/api.utils.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+  const { searchParams } = new URL(request.url);
+
+  return searchParams.get('authUuid') || '';
```
Comment (Taraslysak) [inline app/api/api.utils.ts:4]: `return searchParams.get('authUuid') ?? '';`

File: app/api/users/route.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import { NextResponse } from 'next/server';
+
+import { UsersService } from '@/client';
+import { getAuthUuid } from '../api.utils';
+
+export async function GET(request: Request) {
+  const uuid = getAuthUuid(request);
+
+  try {
+    const users = await UsersService.apiGetUsers(uuid || '');
```
Comment (Taraslysak) [inline app/api/users/route.tsx:10]: you don't need `uuid || ''`

File: services/users.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import axios from 'axios';
+import { Session } from 'next-auth';
+
+import { UserIn } from '@/client';
+
+export const getAllUsers = async (session: Session | null) => {
+  const uuid = session?.user?.uuid || '';
```
Comment (Taraslysak) [inline services/users.ts:7]: replace with util method?

File: services/users.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    const response = await axios.get(`http://localhost:3000/api/users?authUuid=${uuid}`);
+
+    return response.data.users;
+  } catch (error) {
+    throw new Error('Unable to get users.');
+  }
+};
+
+export const createUser = async (session: Session | null, user: UserIn) => {
+  const uuid = session?.user?.uuid || '';
```
Comment (Taraslysak) [inline services/users.ts:19]: replace with util method?

File: services/users.ts [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    const response = await axios.post(`http://localhost:3000/api/users?authUuid=${uuid}`, user);
+
+    return response.data.users;
+  } catch (error) {
+    throw new Error('Unable to create user.');
+  }
+};
+
+export const deleteUser = async (session: Session | null, userUuid: string) => {
+  const authUuid = session?.user?.uuid || '';
```
Comment (Taraslysak) [inline services/users.ts:31]: replace with util method?

**Review by Taraslysak — APPROVED** [review]
Please consider minor refactoring in future PRs

**Pull request merged**